ext/sqlite3/database.c

```diff
@@ -12,6 +12,39 @@
 
 VALUE cSqlite3Database;
 
+static void
+close_or_discard_db(sqlite3RubyPtr ctx)
+{
+    if (ctx->db) {
+        if (ctx->owner == getpid()) {
```

Comment:
NB: on recent `glibc`, `getpid` is no longer cached, so this is an extra syscall. 

But I assume `allocate/close` are not supposed to be hot spots, so it's probably fine.

Reply:
Good to know! I am fully OK with this.